### PR TITLE
feat: [DCS-247] add count_all_space and percent_all_space functions

### PR DESCRIPTION
**Pull request by Ksaurav3380**
## Description

Validity function for count all space and percent all space

## Type of change

Delete irrelevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

- [x] Locally Tested
- [ ] Needs Testing From Production

### CODE REVIEW

**Comment by Ksaurav3380** [comment]
fix: module error

File: dcs_core/core/common/models/validation.py [hunk 1]
```diff
@@ -103,6 +103,10 @@ class ValidationFunction(str, Enum):
     PERCENT_EMPTY_STRING = "percent_empty_string"
     COUNT_NAN = "count_nan"
     PERCENT_NAN = "percent_nan"
+    COUNT_NULL_KEYWORD = "count_null_keyword"
```
Comment (subhankarb) [inline dcs_core/core/common/models/validation.py:106]: This PR did not implemented `COUNT_NULL_KEYWORD`

File: dcs_core/core/validation/completeness_validation.py [hunk 2]
```diff
@@ -89,3 +89,47 @@ def _generate_metric_value(self, **kwargs) -> Union[float, int]:
             )
         else:
             raise ValueError("Invalid data source type")
+
+
+class CountEmptyStringValidation(Validation):
```
Comment (subhankarb) [inline dcs_core/core/validation/completeness_validation.py:94]: This class has not been used. We can drop the class.

File: dcs_core/core/datasource/sql_datasource.py [hunk 3]
```diff
@@ -674,3 +674,28 @@ def query_negative_metric(
 
         result = self.fetchone(query)[0]
         return round(result, 2) if operation == "percent" else result
+
+    def query_get_all_space_count(
+        self, table: str, field: str, operation: str, filters: str = None
+    ) -> int:
```
Comment (subhankarb) [inline dcs_core/core/datasource/sql_datasource.py:680]: This function can return both int and float

File: dcs_core/core/validation/completeness_validation.py [hunk 4]
```diff
@@ -89,3 +89,47 @@ def _generate_metric_value(self, **kwargs) -> Union[float, int]:
...
+    def _generate_metric_value(self, **kwargs) -> Union[float, int]:
+        if isinstance(self.data_source, SQLDataSource):
+            return self.data_source.query_get_empty_string_count(
+                table=self.dataset_name,
+                field=self.field_name,
+                filters=self.where_filter if self.where_filter is not None else None,
+            )
+        elif isinstance(self.data_source, SearchIndexDataSource):
+            return self.data_source.query_get_empty_string_count(
```
Comment (subhankarb) [inline dcs_core/core/validation/completeness_validation.py:103]: This has not been implemented for search data source. We should use `notImplementedError`

**Review by subhankarb — CHANGES_REQUESTED** [review]

**Review by subhankarb — APPROVED** [review]

**Pull request merged**